src/worker/clientcontrol.js

```diff
@@ -432,20 +434,52 @@ commands.SETPASS = async function(input, con, msg) {
 };
 
 commands.ADDTOKEN = async function(input, con, msg) {
+    const parts = input.split(' ');
+    const duration = parts[0] === '0' ? 0 : ParseDuration(parts[0], 'sec');
+    const comment = duration === null ? input : parts.slice(1).join(' ');
+
     try {
-        let token = await con.userDb.generateUserToken(con.state.authUserId);
-        con.writeStatus('Created new token for your account. You can use it in place of your password: ' + token);
+        let token = await con.userDb.generateUserToken(con.state.authUserId, duration, comment, con.state.host);
+        con.writeStatusWithTags(
+            'Created new token for your account. You can use it in place of your password: ' + token,
+            { '+auth_token': token }
+        );
     } catch (err) {
         l.error('Error creating user token:', err.message);
         con.writeStatus('There was an error creating a new token for your account');
     }
 };
 
+commands.CHANGETOKEN = async function(input, con, msg) {
+    const parts = input.split(' ');
+    if (parts.length !== 2) {
+        con.writeStatus('Usage: changetoken <token> <expires>');
```

Comment:
This could use the same logic as `addtoken` for consistency. So making `expires` optional and adding `comment` to the end